src/base-app/event.cpp

```diff
@@ -0,0 +1,99 @@
+/*Implemented by Jake*/
+
+#include "event.h"
+
+namespace base{
+    event::event()
+    {
+        mainHost = NULL;
+        hosts = NULL;
+        attendingUsers = NULL;
```

Comment:
You probably don't want to set hosts or attendingUsers to NULL here. This is calling the [constructor](http://doc.qt.io/qt-5/qvector.html#QVector-1) that accepts an integer. It would be much better just to use the default constructor.

Reply:
Same with location and hosts, btw.